apps/studio/components/layouts/TableEditorLayout/TableEditorMenu.tsx

```diff
@@ -96,13 +100,18 @@ const TableEditorMenu = () => {
   )
   const isLocked = protectedSchemas.some((s) => s.id === schema?.id)
 
-  const { data: selectedTable } = useTable(id !== undefined ? Number(id) : undefined)
+  const { data: tableData } = useTableEditorQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+    id,
+  })
 
   useEffect(() => {
+    const selectedTable = getTableLikeFromTableEditor(tableData)
     if (selectedTable) {
       setSelectedSchema(selectedTable.schema)
     }
-  }, [selectedTable])
+  }, [tableData])
```

Comment:
Minor nit: am thinking the dependency array here should be`tableData.id`?

Given that this useEffect is just to set the schema, `tableData` being JSON - this will always trigger this useEffect even if the schema didn't change (e.g if i add a new column - verified locally) Although the consequence isn't anything big